src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Aead.cs

```diff
@@ -63,27 +64,25 @@ internal static partial class AppleCrypto
             fixed (byte* plaintextPtr = plaintext)
             fixed (byte* aadPtr = aad)
             {
-                const int Success = 1;
-                const int AuthTagMismatch = -1;
-                int result = AppleCryptoNative_ChaCha20Poly1305Decrypt(
-                    keyPtr, key.Length,
-                    noncePtr, nonce.Length,
-                    ciphertextPtr, ciphertext.Length,
-                    tagPtr, tag.Length,
-                    plaintextPtr, plaintext.Length,
-                    aadPtr, aad.Length);
-
-                if (result != Success)
+                AppleCryptoNative_ChaCha20Poly1305Decrypt(
+                    new(keyPtr, key.Length),
+                    new(noncePtr, nonce.Length),
+                    new(ciphertextPtr, ciphertext.Length),
+                    new(tagPtr, tag.Length),
+                    new(plaintextPtr, plaintext.Length),
+                    new(aadPtr, aad.Length),
```

Comment:
I don't see how this pattern is permissible any more than `var` on the left side of an assignment and the type isn't on the right. We should get the Libraries style guidelines owners to weight in on this.

Reply:
Yeah I can get rid of the target-typed new.